inventory_management_api/routes/cart.py

```diff
@@ -0,0 +1,51 @@
+from fastapi import APIRouter, Depends, HTTPException
+from models import User, Cart, CartItem, Product
+from authentication import get_current_user
+
+router = APIRouter()
+
+@router.post("/cart/add")
+async def add_to_cart(product_id: int, quantity: int, user: User = Depends(get_current_user)):
+    product = await Product.get(id=product_id)
+    cart, _ = await Cart.get_or_create(user=user)
+    cart_item, created = await CartItem.get_or_create(cart=cart, product=product)
+    
+    if not created:
+        cart_item.quantity += quantity
+    else:
+        cart_item.quantity = quantity
+    
+    cart_item.price = product.new_price
+    await cart_item.save()
+    return {"message": "Item added to cart"}
+
+@router.get("/cart")
+async def view_cart(user: User = Depends(get_current_user)):
+    cart, _ = await Cart.get_or_create(user=user)
+    items = await CartItem.filter(cart=cart).prefetch_related('product')
+    total = sum(item.quantity * item.price for item in items)
+    return {
+        "items": [
+            {
+                "product": item.product.name,
+                "quantity": item.quantity,
+                "price": item.price,
+                "total": item.quantity * item.price
+            } for item in items
+        ],
+        "total": total
+    }
+
+@router.put("/cart/update/{item_id}")
+async def update_cart_item(item_id: int, quantity: int, user: User = Depends(get_current_user)):
+    cart = await Cart.get(user=user)
+    item = await CartItem.get(id=item_id, cart=cart)
+    item.quantity = quantity
+    await item.save()
+    return {"message": "Cart item updated"}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding stock check in update</b></div>
<div id="fix">

The `update_cart_item` function doesn't validate if the new quantity is available in stock. Consider adding a stock check similar to the `add_to_cart` function.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -40,6 +40,9 @@
  async def update_cart_item(item_id: int, quantity: int, user: User = Depends(get_current_user)):
      cart = await Cart.get(user=user)
      item = await CartItem.get(id=item_id, cart=cart)
 +    product = await item.product
 +    if product.stock < quantity:
 +        raise HTTPException(status_code=400, detail="Not enough stock available")
      item.quantity = quantity
      await item.save()
      return {"message": "Cart item updated"}
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/trulyfurqan/eshop-fastapi/pull/8#issuecomment-2472732780>#b8ff8d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged